src/repository.rs

```diff
@@ -537,7 +538,318 @@ where
+        HttpRepositoryBuilder2 {
+            uri: uri,
+            client: client,
+            interchange: PhantomData,
+            user_agent: None,
+            metadata_prefix: None,
+            targets_prefix: None,
+            min_bytes_per_second: 4096,
+        }
+    }
+
+    /// Set the User-Agent prefix.
+    ///
+    /// Callers *should* include a custom User-Agent prefix to help maintainers of TUF repositories
+    /// keep track of which client versions exist in the field.
+    ///
+    pub fn user_agent<T: Into<String>>(mut self, user_agent: T) -> Self {
+        self.user_agent = Some(user_agent.into());
+        self
+    }
+
+    /// The argument `metadata_prefix` is used to provide an alternate path where metadata is
+    /// stored on the repository. If `None`, this defaults to `/`. For example, if there is a TUF
+    /// repository at `https://tuf.example.com/`, but all metadata is stored at `/meta/`, then
+    /// passing the arg `Some("meta".into())` would cause `root.json` to be fetched from
+    /// `https://tuf.example.com/meta/root.json`.
+    pub fn metadata_prefix(mut self, metadata_prefix: Vec<String>) -> Self {
+        self.metadata_prefix = Some(metadata_prefix);
+        self
+    }
+
+    /// The argument `targets_prefix` is used to provide an alternate path where targets is
+    /// stored on the repository. If `None`, this defaults to `/`. For example, if there is a TUF
+    /// repository at `https://tuf.example.com/`, but all targets are stored at `/targets/`, then
+    /// passing the arg `Some("targets".into())` would cause `hello-world` to be fetched from
+    /// `https://tuf.example.com/targets/hello-world`.
+    pub fn targets_prefix(mut self, targets_prefix: Vec<String>) -> Self {
+        self.targets_prefix = Some(targets_prefix);
+        self
+    }
+
+    /// Set the minimum bytes per second for a read to be considered good.
+    pub fn min_bytes_per_second(mut self, min: u32) -> Self {
+        self.min_bytes_per_second = min;
+        self
+    }
+
+    /// Build a `HttpRepository2`.
+    pub fn build(self) -> HttpRepository2<C, D> {
+        let user_agent = match self.user_agent {
+            Some(user_agent) => user_agent,
+            None => "rust-tuf".into(),
+        };
+
+        HttpRepository2 {
+            uri: self.uri,
+            client: self.client,
+            interchange: self.interchange,
+            user_agent: user_agent,
+            metadata_prefix: self.metadata_prefix,
+            targets_prefix: self.targets_prefix,
+            min_bytes_per_second: self.min_bytes_per_second,
+        }
+    }
+}
+
+/// A repository accessible over HTTP.
+pub struct HttpRepository2<C, D>
+where
+    C: Connect + Sync + 'static,
+    D: DataInterchange,
+{
+    uri: Uri,
+    client: Client<C>,
+    user_agent: String,
+    metadata_prefix: Option<Vec<String>>,
+    targets_prefix: Option<Vec<String>>,
+    min_bytes_per_second: u32,
+    interchange: PhantomData<D>,
+}
```

Comment:
As far as I can tell, `HttpRepository.url` only appears in the API in `HttpRepositoryBuilder::new`. So instead of having a whole `HttpRespository2`, couldn't you just have 2 constructors for `HttpRepositoryBuilder`, which internally always converts to `http::Uri`?

Reply:
Done.